File: 0020-valid-parentheses/0020-valid-parentheses.java
import java.util.*;

class Solution {
    public boolean isValid(String s) {
        Deque<Character> stack = new ArrayDeque<>();

        for(int i = 0;  i<s.length(); i++){
            char c = s.charAt(i);
            if(c=='(' || c=='[' || c == '{'){
                stack.push(c);
            } else{
                
                if(stack.size()>0){
                   char peek = stack.pop();
                   if(c==')' && peek =='('){
                     continue;
                   }
                   if(c=='}' && peek =='{'){
                     continue;
                   }
                   if(c==']' && peek =='['){
                     continue;
                   }
                   return false;
                } else{
                    return false;
                }
                
                
            }
        }
        if(stack.size()==0){
            return true;
        } else{
            return false;
        }
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    const stack = [];
    const map = new Map();
    map.set('(',')');
    map.set('{','}');
    map.set('[',']');
    for(let i=0; i<s.length; i++){
        if(s[i]=='(' || s[i]=='{' || s[i]=='['){
            stack.push(s[i]);
        } else{
            if(stack.length<0){
                return false;
            } else{
                const poped = stack.pop();
                if(s[i] == map.get(poped)){
                    continue;
                } else{
                    return false;
                }
            }
        }
    }
    if(stack.length==0){
        return true;
    } else{
        return false;
    }
};